### PR TITLE
log: add btp log parser

### PR DESCRIPTION
**Pull request by piotrnarajowski**
This will generate new log files for each test run with content similar to iutctl logs, but each packet sent/received through BTP will be decoded and presented in readable form.
Here's log snippet: 
image

### CODE REVIEW

**Comment by piotrnarajowski** [comment]
@sjanc please take a look. Now log file is created in same directory as other log files. Also added btp error status values logging 
image

**Comment by Thalley** [comment]
@piotrnarajowski do we need yet another log? Couldn't we just modify the existing log to be like this?

cc @koffes @alexsven

**Comment by piotrnarajowski** [comment]
> @piotrnarajowski do we need yet another log? Couldn't we just modify the existing log to be like this?

I guess we can get rid of existing one and leave decoded one, it's more useful imo

**Comment by Thalley** [comment]
> > @piotrnarajowski do we need yet another log? Couldn't we just modify the existing log to be like this?
> 
> I guess we can get rid of existing one and leave decoded one, it's more useful imo

Agreed

File: autopts/pybtp/defs.py [hunk 1]
```diff
@@ -55,19 +55,20 @@ def BIT(bit):
...
-CORE_LOG_MESSAGE = 0x05
-CORE_EV_IUT_READY = 0x80
-GAP_READ_SUPPORTED_COMMANDS = 0x01
-GAP_READ_CONTROLLER_INDEX_LIST = 0x02
+BTP_CMD_CORE_READ_SUPPORTED_COMMANDS = 0x01
+BTP_CMD_CORE_READ_SUPPORTED_SERVICES = 0x02
+BTP_CMD_CORE_REGISTER_SERVICE = 0x03
+BTP_CMD_CORE_UNREGISTER_SERVICE = 0x04
+BTP_CMD_CORE_LOG_MESSAGE = 0x05
+BTP_EV_CORE_IUT_READY = 0x80
```
Comment (Thalley) [inline autopts/pybtp/defs.py:69]: What is the rationale for `BTP_CMD_<SERVICE>` and `BTP_EV_<SERVICE>` compared to `BTP_<SERVICE>_CMD` and `BTP_<SERVICE>_EV?

File: autopts/pybtp/defs.py [hunk 2]
```diff
@@ -55,19 +55,20 @@ def BIT(bit):
...
+BTP_CMD_GAP_READ_SUPPORTED_COMMANDS = 0x01
+BTP_CMD_GAP_READ_CONTROLLER_INDEX_LIST = 0x02
 GAP_SETTINGS_POWERED = 0
```
Comment (Thalley) [inline autopts/pybtp/defs.py:72]: Is the plan to also modify the remaining ones to be prefixed with BTP?

File: autopts/pybtp/defs.py (hunk 1 above)
Comment (Thalley) [inline autopts/pybtp/defs.py:69]: Suggest to use `EVT` instead of `EV` so that it has the same length as `CMD`

File: autopts/pybtp/defs.py [hunk 3]
```diff
@@ -55,19 +55,20 @@ def BIT(bit):
 BTP_SERVICE_ID_PBP = 0x1e
 # GENERATOR append 1
 
+BTP_ERROR = 0x00
```
Comment (Thalley) [inline autopts/pybtp/defs.py:58]: This seems unused, also defining `ERROR` to `0x00` seems quite atypical as `0` is more often than not used for success.

File: autopts/pybtp/defs.py (hunk 2 above)
Comment (piotrnarajowski) [reply autopts/pybtp/defs.py:72]: Yes, I have finished modifying all services  and commands locally and will push changes in a matter of minutes

File: autopts/pybtp/defs.py (hunk 3 above)
Comment (piotrnarajowski) [reply autopts/pybtp/defs.py:58]: "As general rule of thumb, the opcode for command matches the opcode for a response. Or the opcode 0x00 for an error is returned"
Please see overview.txt
But you are correct that this variable is unused

Comment (Thalley) [reply autopts/pybtp/defs.py:58]: Ah, right. Saw this the other day in the logs as well. If kept, it should be rename to something like `BTP_ERROR_OPCODE` as `BTP_ERROR` is very generic

File: autopts/pybtp/defs.py (hunk 1 above)
Comment (piotrnarajowski) [reply autopts/pybtp/defs.py:69]: Decided to stick with EV convention as that's the way they are named both in mynewt and zephyr

**Comment by codecoup-tester** [comment]
Scheduled PR https://github.com/auto-pts/auto-pts/pull/1277#issuecomment-2535923231, board: nrf52, estimated start time: 13:59:37, test case count: 2, estimated duration: 0:00:28<details><summary>Test cases to be run</summary>GATT/CL/GAC/BV-01-C<br>GAP/BROB/BCST/BV-01-C</details>

**Comment by piotrnarajowski** [comment]
#AutoPTS run zephyr BAP BASS CCP CSIP MCP MICP TMAP VCP --test_case_limit 1

**Comment by codecoup-tester** [comment]
Scheduled PR https://github.com/auto-pts/auto-pts/pull/1277#issuecomment-2535927650, board: nrf53, estimated start time: 14:02:09, test case count: 8, estimated duration: 0:03:22<details><summary>Test cases to be run</summary>BAP/CL/CGGIT/SER/BV-01-C<br>BASS/SR/SGGIT/SER/BV-01-C<br>CCP/CL/CGGIT/SER/BV-01-C<br>CSIP/CL/CGGIT/SER/BV-01-C<br>MCP/CL/CGGIT/SER/BV-02-C<br>MICP/CL/CGGIT/SER/BV-01-C<br>TMAP/SR/SGGIT/SER/BV-01-C<br>VCP/VC/CGGIT/SER/BV-01-C</details>

**Comment by codecoup-tester** [comment]
AutoPTS Bot results:
No failed test found.
<details><summary>Successful tests (2)</summary>GAP     GAP/BROB/BCST/BV-01-C           PASS<br>GATT    GATT/CL/GAC/BV-01-C             PASS</details>

**Comment by piotrnarajowski** [comment]
#AutoPTS run zephyr MICP/CL/CGGIT/SER/BV-01-C

**Comment by codecoup-tester** [comment]
Scheduled PR https://github.com/auto-pts/auto-pts/pull/1277#issuecomment-2536023302, board: nrf53, estimated start time: 14:42:19, test case count: 1, estimated duration: 0:00:19<details><summary>Test cases to be run</summary>MICP/CL/CGGIT/SER/BV-01-C</details>

**Comment by codecoup-tester** [comment]
AutoPTS Bot results:
No failed test found.
<details><summary>Successful tests (1)</summary>MICP    MICP/CL/CGGIT/SER/BV-01-C       PASS</details>

File: autopts/pybtp/defs.py (hunk 2 above)
Comment (Thalley) [reply autopts/pybtp/defs.py:72]: Was referring to values such as `GAP_SETTINGS_POWERED` which do not have the BTP prefix

Comment (piotrnarajowski) [reply autopts/pybtp/defs.py:72]: For now only commands and events are refactored as this makes  btp header decoding easier.

**Review by sjanc — APPROVED** [review]
since this is very prone to conflicts, lets merge this as is now,   some fine tuning can be done later on if needed

File: autopts/pybtp/defs.py (hunk 1 above)
Comment (sjanc) [reply autopts/pybtp/defs.py:69]: actually,  I just notice this comment and TBH I think we should do BTP_<SERVICE>_CMD and BTP_<SERVICE>_EV

**Review by sjanc — CHANGES_REQUESTED** [review]
lets use 'BTP_<SERVICE>_CMD' pattern

**Comment by piotrnarajowski** [comment]
@sjanc @Thalley changed naming convention to BTP_<SERVICE>_CMD and BTP_<SERVICE>_EV

**Review by Thalley — APPROVED** [review]

**Review by sjanc — APPROVED** [review]

**Pull request merged**